gtsam/hybrid/tests/testGaussianMixture.cpp

```diff
@@ -79,8 +80,8 @@ TEST(GaussianMixture, GaussianMixtureModel) {
   double midway = mu1 - mu0;
   auto eliminationResult =
       gmm.toFactorGraph({{Z(0), Vector1(midway)}}).eliminateSequential();
-  auto pMid = *eliminationResult->at(0)->asDiscrete();
-  EXPECT(assert_equal(DiscreteConditional(m, "60/40"), pMid));
+  auto pMid = eliminationResult->at(0)->asDiscrete<TableDistribution>();
+  EXPECT(assert_equal(TableDistribution(m, "60 40"), *pMid));
```

Comment:
I'm still worried that we now changed the entire hybrid inference pipeline instead of  just the use case we wanted to optimize. In particular, we now silently impose the 32 variable in the last clique constraint on everyone. This is why I was wondering about a more "bespoke" flag as to when to use the TableDistribution. Let;s brainstorm?